src/components/PlayerSubmissionForm.js

```diff
@@ -1,24 +1,106 @@
 import React, { useState } from 'react';
 import PropTypes from 'prop-types';
-
 import './PlayerSubmissionForm.css';
 
-const PlayerSubmissionForm = () => {
+const PlayerSubmissionForm = (props) => {
+  const setInitialState = () => {
+    const initialStateValue = {}; //this is an object with an ability to set key like hash
+    for ( const field of props.fields ) {
+      if (field.key) {// if this is true
+        initialStateValue[field.key] = ''; //set key and value
+      }
+    }
+    return initialStateValue;
+
+  }
+
+  const [words, updateWords] = useState(setInitialState());
+
+  /*event handlers/listeners*/
+  //this is how the form is gathering and building up data for form
+  const onInputChange = (event) => {
+  
+    const newPlayerSubmission = {
+      ...words 
+    };
+
+    const { name, value } =  event.target
+
+    // This sets newformFields to the old value of the form state
+    // and then updates the one field that changed.
+    newPlayerSubmission[name] = value; // updated the value using the hash key
+    updateWords(newPlayerSubmission);  // updated the form with new values
+  }
+
+  //This is the finished data being passed to game
+  const onFormWordSubmission = (event) => {
+    // prevent the form from being submitted
+    event.preventDefault();
+    // print user data
+    console.log(words);
+    // pass data to Game-callback function-your submission is ready
+    props.sendSubmission(words)
+    // clear the submission, get ready for the next player
+    updateWords(setInitialState()); 
+  };
+
   return (
     <div className="PlayerSubmissionForm">
-      <h3>Player Submission Form for Player #{  }</h3>
-
-      <form className="PlayerSubmissionForm__form" >
+      <h3>Player Submission Form for Player # { props.index }</h3>
```

Comment:
This will cause all the tests in `Game.test.js` to pass.

```suggestion
      <h3>Player Submission Form for Player #{ props.index }</h3>
```